game-backend/src/helper/powerup.ts

```diff
@@ -0,0 +1,19 @@
+import {
+  State,
+  Consumable,
+  GameObjectTypes,
+  Consume,
+  GameObject,
+} from '@darwin/types';
+import powerupMap from '../game-engine/mechanics/powerup-spawner/powerups/index';
+
+export const filterPowerups = (objects: GameObject[]): Consumable[] =>
+  objects.filter(
+    obj => !!(powerupMap as Record<GameObjectTypes, Consume>)[obj.type]
+  ) as Consumable[];
```

Comment:
Try not to type cast - this erases the benefit of TypeScript. This is type cast sorcery 😅